File: main2(version1)/treescore.m
function [F] = treescore()
clear all

objective = @(x) myObjective(x);
x0 = ones(1,9);
lb = [zeros(1,6),ones(1,3)*-10];
ub = [ones(1,6),ones(1,3)*10];

options = optimoptions('fmincon', 'Display', 'off','MaxFunctionEvaluations',50000,'MaxIterations',10000);
[x_optimal,fval,exitflag] = fmincon(objective, x0, [], [], [], [], lb, ub, [], options);

disp('Optimal solution t:');
disp(-log(x_optimal(1:6)));
disp('Optimal solution c:');
disp(exp(x_optimal(7:9)));
disp('Optimal objective function value:');
disp(exp(vpa(-fval)));
F= -fval;

end

function result = myObjective(x)
	variables
	lambda_formula
	lambda_formula_trivial
	suml=1-exp(c1)*log(t1)-exp(c2)*log(t2)-exp(c3)*log(t3)-sum(la);
	F = sum(log(l))-length(l)*log(suml);
	result = -F;
end

File: main2(version2)/treescore.cpp
#include <iostream>
#include <vector>
#include "optimizer.h"
#include "variables.h"
#include "lambda_formula.h"
#include "lambda_formula_trivial.h"

int main() {
    int n_vars = 9;
    std::vector<double> x0(n_vars, 1.0);
    std::vector<double> lb(n_vars, 0.0);
    for (int i = 6; i < n_vars; ++i) lb[i] = -10;
    std::vector<double> ub(n_vars, 1.0);
    for (int i = 6; i < n_vars; ++i) ub[i] = 10;

    Optimizer optimizer;
    std::vector<double> x_optimal;
    double fval;
    optimizer.optimize(x0, lb, ub, x_optimal, fval);

    std::cout << "Optimal solution:" << std::endl;
    for (size_t i = 0; i < x_optimal.size(); ++i) {
        std::cout << "x[" << i + 1 << "] = " << x_optimal[i] << std::endl;
    }
    std::cout << "Optimal objective function value: " << fval << std::endl;
    return 0;
}
